docs/howto/krustlet-on-kubernetes-with-inlets.md

```diff
@@ -0,0 +1,273 @@
+
+```yaml
+$ cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: inlets
+  labels:
+    app: inlets
+spec:
+  type: ClusterIP
+  ports:
+    - port: 8000
+      protocol: TCP
+      targetPort: 8000
+      name: control
+    - port: 3000
+      protocol: TCP
+      targetPort: 3000
+      name: data
+  selector:
+    app: inlets
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inlets
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: inlets
+  template:
+    metadata:
+      labels:
+        app: inlets
+    spec:
+      volumes:
+        - name: ghosttunnel-tls-volume
+          secret:
+            secretName: ghosttunnel-tls
+        - name: inlets-token-volume
+          secret:
+            secretName: inlets-token
+      containers:
+      - name: inlets
+        image: inlets/inlets:2.7.0
+        imagePullPolicy: Always
+        command: ["inlets"]
+        args:
+        - "server"
+        - "--token-from=/var/inlets/token"
+        - "--control-port=8000"
+        - "--port=3001"
+        volumeMounts:
+          - name: inlets-token-volume
+            mountPath: /var/inlets/
+      - name: ghosttunnel
+        image: squareup/ghostunnel:v1.5.2
+        imagePullPolicy: Always
+        args:
+        - "server"
+        - "--target=127.0.0.1:3001"
+        - "--listen=0.0.0.0:3000"
+        - "--cert=/etc/tls/tls.crt"
+        - "--key=/etc/tls/tls.key"
+        - "--disable-authentication"
+        volumeMounts:
+          - name: ghosttunnel-tls-volume
+            mountPath: /etc/tls
+EOF
+```
+
+* Port-forward or expose the inlets server
+
+```shell
+kubectl port-forward svc/inlets 8000:8000 &
+```
+
+You can also expose inlets via Ingress using cert-manager to give its control-port a TLS certificate
+
+* Run the `inlets client` on your computer
+
+```shell
+inlets client \
+  --upstream https://127.0.0.1:3000 \
+  --remote ws://127.0.0.1:8000 --token $(token.txt)
+```
+
+* Get the inlets server's service IP
+
+```shell
+export NODE_IP=$(kubectl get service inlets -o jsonpath="{.spec.clusterIP}")
+```
+
+* Run the krustlet now
+
+```shell
+krustlet-wasi --node-ip $NODE_IP --pfx-password password
+```
+
+* Show that the krustlet node has joined the cluster
+
+```shell
+$ kubectl get nodes -o wide
+NAME                   STATUS   ROLES    AGE    VERSION   INTERNAL-IP      EXTERNAL-IP       OS-IMAGE                       KERNEL-VERSION         CONTAINER-RUNTIME
+pool-3xbltttyc-3no2p   Ready    <none>   153m   v1.16.6   10.131.25.141    206.189.19.185    Debian GNU/Linux 9 (stretch)   4.19.0-0.bpo.6-amd64   docker://18.9.2
+pool-3xbltttyc-3no2s   Ready    <none>   153m   v1.16.6   10.131.28.223    206.189.123.184   Debian GNU/Linux 9 (stretch)   4.19.0-0.bpo.6-amd64   docker://18.9.2
+krustlet               Ready    agent    43m    v1.17.0   10.245.157.226   <none>            <unknown>                      <unknown>              mvp
+```
+
+## Step 4: Test that things work
```

Comment:
Can you remove this section? This is described in the [quickstart](https://github.com/deislabs/krustlet/blob/master/docs/intro/quickstart.md). We're doing this to prevent having to update the examples in multiple locations across the documentation.

Reply:
It's actually in the KinD guide so I copied it for consistency. Happy to remove it from both if that's preferable?

Reply:
(From the AKS guide)

Reply:
Oh! Good point. Let's keep this PR constrained to this guide. I'll go ahead and remove that section from the KinD howto in another PR.